test/cross/cross_test.go

```diff
@@ -0,0 +1,201 @@
+						"simple_start": {
+							Number: 1,
+						},
+						"simple_mid": {
+							Number: 1,
+						},
+					},
+				},
+				"callee": {
+					Status: serverlessv1alpha1.InstanceStatus{
+						HostIP: "127.0.0.1",
+						PodIP:  "127.0.0.1:9090",
+					},
+					ProcessRuntimes: serverlessv1alpha1.ProcessRuntimes{
+						"simple_branch_1": {
+							Number: 1,
+						},
+						"simple_branch_2": {
+							Number: 1,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func GetSampleWorkflow() *serverlessv1alpha1.Workflow {
+	return &serverlessv1alpha1.Workflow{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: APIVersion,
+			Kind:       WorkflowKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "simple",
+			Namespace: "default",
+		},
+		Spec: serverlessv1alpha1.WorkflowSpec{
+			Spec: []serverlessv1alpha1.Flow{
+				{
+					Name:       "simple_start",
+					Function:   "simple_start",
+					Outputs:    []string{"simple_mid"},
+					Conditions: []*serverlessv1alpha1.Condition{},
+					Statement:  serverlessv1alpha1.Direct,
+					Role:       serverlessv1alpha1.Start,
+				},
+				{
+					Name:       "simple_mid",
+					Function:   "simple_mid",
+					Outputs:    []string{"simple_branch_1", "simple_branch_2"},
+					Conditions: []*serverlessv1alpha1.Condition{},
+					Statement:  serverlessv1alpha1.Direct,
+					Role:       "",
+				},
+				{
+					Name:       "simple_branch_1",
+					Function:   "simple_branch_1",
+					Outputs:    []string{},
+					Conditions: []*serverlessv1alpha1.Condition{},
+					Statement:  serverlessv1alpha1.Direct,
+					Role:       serverlessv1alpha1.End,
+				},
+				{
+					Name:       "simple_branch_2",
+					Function:   "simple_branch_2",
+					Outputs:    []string{},
+					Conditions: []*serverlessv1alpha1.Condition{},
+					Statement:  serverlessv1alpha1.Direct,
+					Role:       serverlessv1alpha1.End,
+				},
+			},
+		},
+		Status: serverlessv1alpha1.WorkflowStatus{},
+	}
+}
+
+func DumpConfig(object runtime.Object, fileName string) error {
+	scheme := runtime.NewScheme()
+	_ = serverlessv1alpha1.AddToScheme(scheme)
+	serializer := json.NewSerializerWithOptions(yaml.DefaultMetaFactory, scheme, scheme, json.SerializerOptions{true, false, false})
+	dumpFile, err := os.OpenFile(fileName, os.O_WRONLY|os.O_CREATE, 0)
+	if err != nil {
+		return err
+	}
+
+	err = serializer.Encode(object, dumpFile)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func stdOutDump(cmd *exec.Cmd, fileName string) {
+	// open the out file for writing
+	outfile, err := os.Create(fileName)
+	if err != nil {
+		panic(err)
+	}
+	defer outfile.Close()
+
+	cmd.Stdout = outfile
+
+	err = cmd.Start()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func TestDump(t *testing.T) {
+	err := DumpConfig(GetSampleWorkflow(), "./config/workflow.yaml")
+	t.Log(err)
+	err = DumpConfig(GetSampleWorkflowRuntime(), "./config/workflowruntime.yaml")
+	t.Log(err)
+}
+
+func TestCross(t *testing.T) {
+	Convey("test across local scheduler test", t, func() {
```

Comment:
so it tests that the data can pass through by network. I don't know which param let the local scheduler stop creating a local process first. I need a hint

Reply:
the flag `-i` means use static  middleware, which has a higher order than `lsds`, so it can send http request direct without enter `lsds` to create `Source` event.

Reply:
I will add comment at here about flags